packages/liquidator/src/liquidator.js

```diff
@@ -118,18 +121,53 @@ class Liquidator {
           // Override must be one of the default logging levels: ['error','warn','info','http','verbose','debug','silly']
           return Object.values(overrides).every(param => Object.keys(this.logger.levels).includes(param));
         }
+      },
+      whaleDefenseFundWei: {
+        // by default make this disabled
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return this.toBN(x).gte(this.toBN("0"));
+        }
+      },
+      defenseActivationPercent: {
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return parseFloat(x) >= 0 && parseFloat(x) <= 100;
+        }
       }
     };
 
     // Validate and set config settings to class state.
-    Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));
+    const configWithDefaults = createObjectFromDefaultProps(config, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // this takes in config, and emits log events
+    this.liquidationStrategy = LiquidationStrategy(
+      {
+        ...configWithDefaults,
+        empMinSponsorSize: empProps.minSponsorSize,
+        withdrawalLiveness: empProps.liquidationLiveness
+      },
+      this.web3.utils,
+      (type, severity, data = {}) => {
+        if (type !== "log") return;
+        if (this.logger[severity] == null) throw new Error("Unknown logger severity: " + severity);
```

Comment:
One catch with this way of validating (and I guess a potential downside with your `emit` pattern above) is that this could fail in the _logger_ even if the behavior is correct due to an invalid `severity` level. This means it's validating the LiquidationStratry implementation at run time. This might not be an issue though.

Perhaps consider also throwing the error values returned, so that the log message will at least contain what was thrown from the `LiquidationStratergy`. For example, modify what was thrown to contain a spread `data` prop.

Reply:
good point, I will refactor this